mbed-hal/spi_api.h

```diff
@@ -32,10 +32,33 @@
 #define SPI_FILL_WORD         (0xFFFFFFFF)
 
 typedef enum {
-    SPI_MSB,
-    SPI_LSB
+    SPI_ORDER_MSB,
+    SPI_ORDER_LSB
 } spi_bitorder_t;
 
+typedef enum {
+    SPI_BITS_4 = 4,
+    SPI_BITS_5 = 5,
+    SPI_BITS_6 = 6,
+    SPI_BITS_7 = 7,
+    SPI_BITS_8 = 8,
+    SPI_BITS_9 = 9,
+    SPI_BITS_10 = 10,
+    SPI_BITS_11 = 11,
+    SPI_BITS_12 = 12,
+    SPI_BITS_13 = 13,
+    SPI_BITS_14 = 14,
+    SPI_BITS_15 = 15,
+    SPI_BITS_16 = 16
```

Comment:
You have to "range check" anyway, because you need to translate the `uint8_t` or enum to the actual peripheral bitmapping anyway, so both are actually completely fine.